lib/Parse/ParsePattern.cpp

```diff
@@ -895,6 +895,8 @@ ParserResult<TypeRepr> Parser::parseThrowsType() {
     
     // we parse every type (including function types)
     ParserResult<TypeRepr> throwsTypeRepr = parseType(diag::expected_parenthesized_type_after_throws);
+    if (throwsTypeRepr.hasCodeCompletion())
+      return makeParserCodeCompletionResult<TypeRepr>();
```

Comment:
This should probably come after the `parseMatchingToken()` call on line 918, because I think, that we still want to match the `)` even after a code completion. Maybe we could set a flag here, that we later want to return a code completion result.